crypto/x509/x509_vfy.c

```diff
@@ -571,6 +572,9 @@ static int check_chain_extensions(X509_STORE_CTX *ctx)
             /* Check sig alg consistency acc. to RFC 5280 section 4.1.1.2 */
             if (X509_ALGOR_cmp(&x->sig_alg, &x->cert_info.signature) != 0)
                 ctx->error = X509_V_ERR_SIGNATURE_ALGORITHM_INCONSISTENCY;
+            /* Check for presence of explicit elliptic curve parameters */
+            if (!check_curve(x))
```

Comment:
Should this have a tri-state (e.g. -1 for malformed/unsupported) like other checks in the chain?

Reply:
Also, should we have an opt-out mechanism for this check in 3.0?

I expect in 1.1.1 we would make this check an opt-in, but I have no idea if we need to provide an opt-out in 3.0 to support existing applications and internal PKIs with certificates encoded with explicit parameters.

I am not requesting a change here, just want to know what @t8m and @DDvO think about it.

Reply:
I also would be in favor of enabling the checks on strict RFC requirements by default.
Yet at least @kroeckx mentioned that this has the potential to disrupt use of many existing ill-formed certs.

Reply:
I think we can actually always check most of the new checks that got added, and not only do it in strict mode. This is one of the checks that can clearly be done in non-strict mode, even for self-signed certificates. I wouldn't mind moving this call to around around line 525.

Reply:
I am not sure it should be applied to self-signed certificates. There is no reason for that as IMO the PKI RFC does not really apply to them. But yes, I have no problem always applying the check in 3.0 and keep it for the strict checks only in 1.1.1.

Reply:
> Should this have a tri-state (e.g. -1 for malformed/unsupported) like other checks in the chain?

The question is what to do for the -1 case, give a different error?

Reply:
Yes.